cluster-autoscaler/cloudprovider/packet/packet_manager_rest.go

```diff
@@ -229,132 +247,182 @@ func randString8() string {
 	return string(b)
 }
 
-func (mgr *packetManagerRest) createNode(cloudinit, nodegroup string) {
+func (mgr *packetManagerRest) createNode(cloudinit, nodegroup string) error {
 	udvars := CloudInitTemplateData{
 		BootstrapTokenID:     os.Getenv("BOOTSTRAP_TOKEN_ID"),
 		BootstrapTokenSecret: os.Getenv("BOOTSTRAP_TOKEN_SECRET"),
 		APIServerEndpoint:    mgr.apiServerEndpoint,
 	}
-	ud := renderTemplate(cloudinit, udvars)
+
+	ud, err := renderTemplate(cloudinit, udvars)
+	if err != nil {
+		return fmt.Errorf("failed to create userdata from template: %w", err)
+	}
+
 	hnvars := HostnameTemplateData{
 		ClusterName: mgr.clusterName,
 		NodeGroup:   nodegroup,
 		RandString8: randString8(),
 	}
-	hn := renderTemplate(mgr.hostnamePattern, hnvars)
+	hn, err := renderTemplate(mgr.hostnamePattern, hnvars)
+	if err != nil {
+		return fmt.Errorf("failed to create hostname from template: %w", err)
+	}
+
+	if err := mgr.createDevice(hn, ud, nodegroup); err != nil {
+		return err
```

Comment:
Ditto annotate errors.
```suggestion
		return fmt.Errorf("failed to create device %q in node group %q: %w", hn, nodegroup, err)
```